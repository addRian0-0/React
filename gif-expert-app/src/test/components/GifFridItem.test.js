import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem"

describe("probando gifFridItem", () => {

    const title = "Un titulo";
    const url = "https://localhost/algo.jsp"
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('should ', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('debe tener un parrafo con el title ', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title)

    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })
    
    test('debe tener la clase de animate css ', () => {
        
        const div = wrapper.find('div');
        expect(div.prop('className')).toBe("card animate__animated animate__fadeIn");

    })  

})