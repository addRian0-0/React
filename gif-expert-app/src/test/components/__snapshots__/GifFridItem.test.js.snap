// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando gifFridItem should  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jsp"
  />
  <p>
    Un titulo
  </p>
</div>
`;
